File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/image/ImageRepository.java
package kyonggiyo.persistence.image;

import kyonggiyo.application.image.domain.entity.Image;
import kyonggiyo.application.image.domain.vo.ImageType;

import java.util.List;

public interface ImageRepository {

    List<Image> saveAll(List<Image> images);

    List<Image> findByImageTypeAndReferenceId(ImageType imageType, Long referenceId);

    List<Image> findByImageTypeAndReferenceIdIn(ImageType imageType, List<Long> ids);

    void deleteById(Long id);

    void deleteAllByIdInBatch(List<Long> ids);

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/image/repository/ImageRepository.kt
package kyonggiyo.image.repository

import kyonggiyo.application.image.domain.entity.Image
import kyonggiyo.application.image.domain.vo.ImageType

interface ImageRepository {

    fun saveAll(images: List<Image>): List<Image>

    fun findByImageTypeAndReferenceId(imageType: ImageType, referenceId: Long): List<Image>

    fun findByImageTypeAndReferenceIdIn(imageType: ImageType, ids: List<Long>): List<Image>

    fun deleteById(id: Long)

    fun deleteAllByIdInBatch(ids: List<Long>)

}
